segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/admin/commands/Command.java

```diff
@@ -0,0 +1,259 @@
+                .builder()
+                .connectString(serviceConfig.getZkURL())
+                .namespace("pravega/" + serviceConfig.getClusterName())
+                .retryPolicy(new ExponentialBackoffRetry(serviceConfig.getZkRetrySleepMs(), serviceConfig.getZkRetryCount()))
+                .sessionTimeoutMs(serviceConfig.getZkSessionTimeoutMs())
+                .build();
+        zkClient.start();
+        return zkClient;
+    }
+
+    protected void output(String messageTemplate, Object... args) {
+        System.out.println(String.format(messageTemplate, args));
+    }
+
+    protected boolean confirmContinue() {
+        output("Do you want to continue?[yes|no]");
+        Scanner s = new Scanner(System.in);
+        String input = s.nextLine();
+        return input.equals("yes");
+    }
+
+    //endregion
+
+    //region Arguments
+
+    protected void ensureArgCount(int expectedCount) {
+        Preconditions.checkArgument(this.commandArgs.getArgs().size() == expectedCount, "Incorrect argument count.");
+    }
+
+    protected int getIntArg(int index) {
+        return getArg(index, Integer::parseInt);
+    }
+
+    protected long getLongArg(int index) {
+        return getArg(index, Long::parseLong);
+    }
+
+    protected boolean getBooleanArg(int index) {
+        return getArg(index, Boolean::parseBoolean);
+    }
+
+    private <T> T getArg(int index, Function<String, T> converter) {
+        String s = null;
+        try {
+            s = this.commandArgs.getArgs().get(index);
+            return converter.apply(s);
+        } catch (Exception ex) {
+            throw new IllegalArgumentException(String.format("Unexpected argument '%s' at position %d: %s.", s, index, ex.getMessage()));
+        }
+    }
+
+    //endregion
+
+    //region Descriptors
+
+    /**
+     * Describes an argument.
+     */
+    @RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+    @Getter
+    public static class ArgDescriptor {
+        private final String name;
+        private final String description;
+    }
+
+    /**
+     * Describes a Command.
+     */
+    @Getter
+    public static class CommandDescriptor {
+        private final String component;
+        private final String name;
+        private final String description;
+        private final ArgDescriptor[] args;
+        CommandDescriptor(String component, String name, String description, ArgDescriptor... args) {
+            this.component = Exceptions.checkNotNullOrEmpty(component, "component");
+            this.name = Exceptions.checkNotNullOrEmpty(name, "name");
+            this.description = Exceptions.checkNotNullOrEmpty(description, "description");
+            this.args = args;
+        }
+    }
+
+    //endregion
+
+    //region Factory
+
+    /**
+     * Helps create new Command instances.
+     */
+    public static class Factory {
+        private static final HashMap<String, HashMap<String, CommandInfo>> COMMANDS = new HashMap<>();
```

Comment:
This should be an ImmutableMap. (It can be initialized by a static method)

Reply:
Ok, but it didn't really make the code look that much better.